### PR TITLE
kubelet/kuberuntime: switch to runc/libct

### PR DESCRIPTION
**Pull request by kolyshkin**
Related to discussion in #128157 - Let's ensure we use ONLY one cgroups library.

runc/libcontainer/cgroups.ParseCgroupFile is a universal function; for cgroup v2 it returns path to unified in a map with "" as a key.

Let's use it here, dropping cgroups dependency entirely.

Amends commit e86d02b60c1 / PR #128473.

/kind cleanup

```release-note
none
```

### CODE REVIEW

**Comment by kolyshkin** [comment]
I don't think the test failure is related.

/retest

File: pkg/kubelet/kuberuntime/kuberuntime_container_linux.go [hunk 1]
```diff
@@ -352,12 +351,12 @@ var swapControllerAvailable = func() bool {
...
 		if isCgroup2UnifiedMode() {
 			// memory.swap.max does not exist in the cgroup root, so we check /sys/fs/cgroup/<SELF>/memory.swap.max
-			_, unified, err := cgroups.ParseCgroupFileUnified("/proc/self/cgroup")
+			cm, err := libcontainercgroups.ParseCgroupFile("/proc/self/cgroup")
 			if err != nil {
 				klog.V(5).ErrorS(fmt.Errorf("failed to parse /proc/self/cgroup: %w", err), warn)
 				return
 			}
-			p = filepath.Join("/sys/fs/cgroup", unified, "memory.swap.max")
+			p = filepath.Join("/sys/fs/cgroup", cm[""], "memory.swap.max")
```
Comment (kannon92) [inline pkg/kubelet/kuberuntime/kuberuntime_container_linux.go:359]: This seems a bit esoteric..

Can you explain why we are grabbing cm[""] here?

Comment (SergeyKanzhelev) [reply pkg/kubelet/kuberuntime/kuberuntime_container_linux.go:359]: (explain by adding a comment into code)

Comment (kolyshkin) [reply pkg/kubelet/kuberuntime/kuberuntime_container_linux.go]: The answer is a click away (in my $EDITOR, at least) -- take a look at the [ParseCgroupFile docs](https://pkg.go.dev/github.com/opencontainers/runc/libcontainer/cgroups#ParseCgroupFile), it says:

> Note that for cgroup v2 unified hierarchy, there are no per-controller cgroup paths, so the resulting map will have a single element where the key is empty string ("") and the value is the cgroup path the <pid> is in.

Nevertheless, I've added the comment.

**Comment by kolyshkin** [comment]
Added a separate second commit which makes use of `sync.OnceValue`.

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Comment by dims** [comment]
/approve
/lgtm

thanks @kolyshkin

**Comment by pacoxu** [comment]
/milestone v1.32
approved before code freeze

**Pull request merged**